lib/MockXMLHttpRequest.js

```diff
@@ -158,40 +159,45 @@ MockXMLHttpRequest.prototype.send = function(data) {
 
     var response = MockXMLHttpRequest.handle(new MockRequest(self));
 
-    if (response && response instanceof MockResponse) {
+    Promise.resolve(response).then(function(response) {
```

Comment:
I think should be used first matched handler, and if it returns no response this is responsibility of developer to write correct one.